core/lib/basic_types/src/basic_fri_types.rs

```diff
@@ -187,6 +203,156 @@ impl TryFrom<i32> for AggregationRound {
     }
 }
 
+/// Wrapper for mapping from protocol version to prover circuits job stats
+#[derive(Debug)]
+pub struct ProtocolVersionedCircuitProverStats {
```

Comment:
Do you actually, really need this? You first collect a list (iterable) of stats. Then you collect it into this struct. But then you only use it to iterate over and report metrics. Could we just use the iterator instead and spare some dozens of LoC?

I may be wrong here, only had a quick look

Reply:
you addressed exactly my point. I see now why it's needed. Let's keep as is